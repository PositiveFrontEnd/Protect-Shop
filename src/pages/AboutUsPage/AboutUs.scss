.aboutus {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  min-width: 320px;
  margin-top: 8px;
  font-size: 42px;

  .aboutus__block {
    display: flex;
    flex-wrap: wrap;
    gap: 10px;
    align-items: center;
    justify-content: center;
    max-width: 1280px;
  }

  .aboutus__h2 {
    display: flex;
    align-items: center;
    height: 57px;
    font-size: 18px;
    font-family: "Noto Sans" sans-serif;
    text-transform: uppercase;
  }

  .aboutus__photo {
    width: 100%;
    max-width: 100%;
    height: auto;
  }

  .aboutus__mainText {
    width: calc(85% - 10px);
    margin-top: 24px;
    margin-bottom: 24px;
    font-size: 14px;
    line-height: 180%;
  }

  .aboutus__text {
    width: calc(85% - 10px); /* 50% ширина з проміжком між елементами */
    max-width: 1280px;
    margin-top: 24px;
    margin-bottom: 40px;
    font-size: 12px;
    line-height: 150%;
  }

  @media (width >= 992px) {
    .aboutus__block {
      display: grid;
      grid-template-rows: repeat(2, 1fr);
      grid-template-columns: repeat(2, 1fr);
      gap: 10px;
      justify-items: center;
      width: 90%;
    }

    .aboutus__h2 {
      color: var(--black);
      font-size: 24px;
      text-transform: uppercase;
    }

    .aboutus__mainText {
      grid-column: span 1;
      gap: 10px;
      width: 100%;
      color: var(--black);
      font-size: 16px;
      line-height: 30px;
    }

    .second p {
      padding-top: 35px;
    }

    .aboutus__photo {
      grid-row: span 1;
      grid-column: span 1;
      width: 100%;
      height: 90%;
      margin-bottom: 60px;
      object-fit: cover;
    }

    .first {
      order: 3;
      margin-top: -125px;
    }

    .second {
      order: 1;
    }

    .third {
      order: 2;
      margin-top: 125px;
    }

    .fourth {
      order: 4;
      margin-left: 20px;
    }
  }
}
