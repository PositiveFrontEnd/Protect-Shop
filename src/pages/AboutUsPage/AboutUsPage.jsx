import React from "react";
import "./AboutUs.scss";
import aboutUsMan from "/Images/aboutUsMan.jpg";
import aboutUsWoman from "/Images/aboutUsWoman.jpg";

const AboutUsPage = () => {
  return (
    <div className="aboutus">
      <h2 className="aboutus__h2">About us</h2>
      <div className="aboutus__block">
        <img src={aboutUsMan} alt="man" className="aboutus__photo first" />
        <p className="aboutus__mainText second">
          Protect was founded in Kharkiv to meet
          the needs of customers in reliable and high-quality bags and
          backpacks. The founders of the store were students who were constantly
          searching for convenient and practical bags and backpacks for their
          educational needs. <br /><br />{" "}
          <span>
            In 2002, they decided to create their
            own style for bag and backpack store to give customers the maximum
            choice and quality. The main value of the store is satisfaction of
            customers' needs and satisfaction after each purchase. Protect tries
            to always keep up to date with the latest trends and consumer needs
            to provide them with modern and high-quality products.
          </span>
        </p>
        <img src={aboutUsWoman} alt="woman" className="aboutus__photo third" />
        <p className="aboutus__mainText fourth">
          Protect offers a wide range of bags and
          rucksacks, including a variety of models and sizes, all of which can
          be expected for maximum performance and quality. All products offered
          by the seller are tested and tested to ensure compliance with safety
          and quality standards. <br /><br /> Protect
          always strives for improvement and development to provide the best
          possible services and products for its customers. Their team consists
          of sales and customer service experts who are always ready to help you
          with any questions about the products or services provided by the
          store.
        </p>
      </div>
    </div>
  );
};

export default AboutUsPage;

