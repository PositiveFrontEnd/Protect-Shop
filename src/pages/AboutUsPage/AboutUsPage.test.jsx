import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import AboutUsPage from "./AboutUsPage";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("Тестування компонента  AboutUsPage", () => {
    test("Знімок", () => {
        const mockStore = configureStore([thunk]);
        const initialState = {};
        const store = mockStore(initialState);
        const aboutUsPage = render(
            <Provider store={store}>
                <BrowserRouter>
                    <AboutUsPage />
                </BrowserRouter>
            </Provider>
        );

        expect(aboutUsPage).toMatchSnapshot();
    });
});
