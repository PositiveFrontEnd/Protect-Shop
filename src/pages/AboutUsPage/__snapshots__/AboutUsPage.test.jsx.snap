// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  AboutUsPage > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="aboutus"
      >
        <h2
          class="aboutus__h2"
        >
          About us
        </h2>
        <div
          class="aboutus__block"
        >
          <img
            alt="man"
            class="aboutus__photo first"
            src="/Images/aboutUsMan.jpg"
          />
          <p
            class="aboutus__mainText second"
          >
            Protect was founded in Kharkiv to meet the needs of customers in reliable and high-quality bags and backpacks. The founders of the store were students who were constantly searching for convenient and practical bags and backpacks for their educational needs. 
            <br />
            <br />
             
            <span>
              In 2002, they decided to create their own style for bag and backpack store to give customers the maximum choice and quality. The main value of the store is satisfaction of customers' needs and satisfaction after each purchase. Protect tries to always keep up to date with the latest trends and consumer needs to provide them with modern and high-quality products.
            </span>
          </p>
          <img
            alt="woman"
            class="aboutus__photo third"
            src="/Images/aboutUsWoman.jpg"
          />
          <p
            class="aboutus__mainText fourth"
          >
            Protect offers a wide range of bags and rucksacks, including a variety of models and sizes, all of which can be expected for maximum performance and quality. All products offered by the seller are tested and tested to ensure compliance with safety and quality standards. 
            <br />
            <br />
             Protect always strives for improvement and development to provide the best possible services and products for its customers. Their team consists of sales and customer service experts who are always ready to help you with any questions about the products or services provided by the store.
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="aboutus"
    >
      <h2
        class="aboutus__h2"
      >
        About us
      </h2>
      <div
        class="aboutus__block"
      >
        <img
          alt="man"
          class="aboutus__photo first"
          src="/Images/aboutUsMan.jpg"
        />
        <p
          class="aboutus__mainText second"
        >
          Protect was founded in Kharkiv to meet the needs of customers in reliable and high-quality bags and backpacks. The founders of the store were students who were constantly searching for convenient and practical bags and backpacks for their educational needs. 
          <br />
          <br />
           
          <span>
            In 2002, they decided to create their own style for bag and backpack store to give customers the maximum choice and quality. The main value of the store is satisfaction of customers' needs and satisfaction after each purchase. Protect tries to always keep up to date with the latest trends and consumer needs to provide them with modern and high-quality products.
          </span>
        </p>
        <img
          alt="woman"
          class="aboutus__photo third"
          src="/Images/aboutUsWoman.jpg"
        />
        <p
          class="aboutus__mainText fourth"
        >
          Protect offers a wide range of bags and rucksacks, including a variety of models and sizes, all of which can be expected for maximum performance and quality. All products offered by the seller are tested and tested to ensure compliance with safety and quality standards. 
          <br />
          <br />
           Protect always strives for improvement and development to provide the best possible services and products for its customers. Their team consists of sales and customer service experts who are always ready to help you with any questions about the products or services provided by the store.
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
